browser/js/dashboard/gridbox/dashboard.factory.js

```diff
@@ -7,7 +7,15 @@ app.factory('DashboardFactory', function($http, $q, GeneratorFactory){
         }
         return $http.get(link)
         .then(getData)
-        .catch(() => [])
+        .then(function(data){
+            let x = obj.findDataToGraph(data);
+            if (x) {
+                return x;
+            }else{
+                return [data]
+            }
+        })
+        .catch(res => [])
```

Comment:
You might want to also notify the user somehow that data could not be located. Maybe consider using something like [toast](https://github.com/Foxandxss/angular-toastr) or [growl](http://janstevens.github.io/angular-growl-2/) messages.